frontend/js/src/settings/notification-settings/NotificationSettings.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import { toast } from "react-toastify";
+import { Helmet } from "react-helmet";
+
+import Switch from "../../components/Switch";
+
+export default function NotificationSettings() {
+  const [notificationsEnabled, setNotificationsEnabled] = React.useState(false);
+  const [digestEnabled, setDigestEnabled] = React.useState(false);
+  const [digestAge, setDigestAge] = React.useState<number>(7);
+  const [initialDigestEnabled, setInitialDigestEnabled] = React.useState(false);
+  const [initialDigestAge, setInitialDigestAge] = React.useState<number | null>(
+    null
+  );
+  const [loading, setLoading] = React.useState(true);
+  const [saving, setSaving] = React.useState(false);
+
+  React.useEffect(() => {
+    async function fetchDigestSettings() {
+      try {
+        const response = await fetch("/settings/digest-setting/");
```

Comment:
While this is correct and works, how we usually do this initial data fetching is with a react-router RouteLoader utility that automatically hits a POST endpoint on the server of the same name as the page. See for example:
https://github.com/metabrainz/listenbrainz-server/blob/c882a9594398d52f6b77b7b5ca64eaaeb0e6a3cc/frontend/js/src/settings/routes/index.tsx#L34-L42

And that way, you can use the initial data with a hook, and as a bonus you won't need to store the 'initial...' values in the state as they will be present in the loaderData for you to compare with:
https://github.com/metabrainz/listenbrainz-server/blob/c882a9594398d52f6b77b7b5ca64eaaeb0e6a3cc/frontend/js/src/settings/music-services/details/MusicServices.tsx#L44-L50

Reply:
Adding the server side of this example for reference:
https://github.com/metabrainz/listenbrainz-server/blob/1235c47ed65db9d4ba93fd06f1b0989bb5213af6/listenbrainz/webserver/views/settings.py#L192-L254